pkgs/swift2objc/lib/src/generate_wrapper.dart

```diff
@@ -0,0 +1,58 @@
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import 'config.dart';
+import 'generator/generator.dart';
+import 'parser/parser.dart';
+import 'transformer/transform.dart';
+
+Future<void> generateWrapper(Config config) async {
+  final parentDir = Directory.fromUri(config.tempDir ?? defaultTempDir);
+  final tempDir = await parentDir.createTemp(defaultTempDirPrefix);
+
+  await _generateSymbolgraphJson(
+    config.inputFiles,
+    tempDir,
+    config.moduleName,
+  );
+
+  final symbolgraphFileName = '${config.moduleName}$symbolgraphFileSuffix';
+  final symbolgraphJsonPath = path.join(tempDir.path, symbolgraphFileName);
+
+  final declarations = parseAst(symbolgraphJsonPath);
+  final transformedDeclarations = transform(declarations);
+  final wrapperCode = generate(transformedDeclarations);
+
+  File.fromUri(config.outputFile).writeAsStringSync(wrapperCode);
+
+  if (config.deleteTempAfterDone) {
+    tempDir.deleteSync(recursive: true);
+  }
+}
+
+Future<void> _generateSymbolgraphJson(
+  List<Uri> inputFiles,
+  Directory outputDirectory,
+  String moduleName,
+) async {
+  final result = await Process.run(
+    'swiftc',
+    [
+      ...inputFiles.map((uri) => uri.path),
+      '-emit-module',
+      '-emit-symbol-graph',
+      '-emit-symbol-graph-dir',
+      '.',
+      '-module-name',
+      moduleName
+    ],
+    workingDirectory: outputDirectory.path,
+  );
+
+  if (result.exitCode != 0) {
+    throw Exception(
```

Comment:
Try to avoid throwing general `Exception`s. There's usually a more specific exception that will work better. In this case `ProcessException` would be more useful, since it displays the command that was executed (handy for debugging). You'll need to pull the args array out into a variable so you can pass it to the `ProcessException`. You can stick the stdout/stderr in the message arg.

Reply:
Done.
I created similar config classes to the ones u made in swiftgen, including a `Command` class, so I have the executable & the args in it.